### ISSUE TITLE
"Too many disconnects from streaming API" error

### ISSUE THREAD

**Issue by TesLocker** [teslamate-org/teslamate#3721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When the vehicle is parked, TeslaMate repeatedly tries to reconnect to the streaming API.  Once the vehicle starts up, it still is not able to reconnect to the streaming API for the first 11 minutes of the drive.

I see this fairly consistently, even when the car just stops for 5 minutes in the driveway.  As a result, the location tracking is not accurate (updated every 15 seconds via polling), which is not helpful for closing the garage door on departure.

### Expected Behavior

TeslaMate does not connect to streaming API while vehicle is parked.  When the vehicle starts up, TeslaMate connects to the streaming API.

### Steps To Reproduce

Drive vehicle for 5 minutes.
Stop or park vehicle for 5 minutes.
Start driving again and note that the streaming API is not connected. 

### Relevant log output

```shell
2024-03-05 18:22:14.443 [warning] Too many disconnects from streaming API
2024-03-05 18:22:14.443 car_id=1 [info] Creating new connection … 
2024-03-05 18:22:14.444 car_id=1 [info] Disconnecting ...
2024-03-05 18:22:14.465 car_id=1 [info] Connecting ...
2024-03-05 18:28:49.691 [warning] Too many disconnects from streaming API
2024-03-05 18:28:49.691 car_id=1 [info] Creating new connection … 
2024-03-05 18:28:49.691 car_id=1 [info] Disconnecting ...
2024-03-05 18:28:49.706 car_id=1 [info] Connecting ...
2024-03-05 18:35:15.673 [warning] Too many disconnects from streaming API
2024-03-05 18:35:15.673 car_id=1 [info] Creating new connection … 
2024-03-05 18:35:15.673 car_id=1 [info] Disconnecting ...
2024-03-05 18:35:15.685 car_id=1 [info] Connecting ...
2024-03-05 18:41:51.657 [warning] Too many disconnects from streaming API
2024-03-05 18:41:51.658 car_id=1 [info] Creating new connection … 
2024-03-05 18:41:51.658 car_id=1 [info] Disconnecting ...
2024-03-05 18:41:51.670 car_id=1 [info] Connecting ...
2024-03-05 18:48:20.393 [warning] Too many disconnects from streaming API
2024-03-05 18:48:20.393 car_id=1 [info] Creating new connection … 
2024-03-05 18:48:20.393 car_id=1 [info] Disconnecting ...
2024-03-05 18:48:20.413 car_id=1 [info] Connecting ...
2024-03-05 18:54:47.482 [warning] Too many disconnects from streaming API
2024-03-05 18:54:47.482 car_id=1 [info] Creating new connection … 
2024-03-05 18:54:47.482 car_id=1 [info] Disconnecting ...
2024-03-05 18:54:47.493 car_id=1 [info] Connecting ...
2024-03-05 19:01:12.791 [warning] Too many disconnects from streaming API
2024-03-05 19:01:12.791 car_id=1 [info] Creating new connection … 
2024-03-05 19:01:12.791 car_id=1 [info] Disconnecting ...
2024-03-05 19:01:12.805 car_id=1 [info] Connecting ...
2024-03-05 19:07:44.755 [warning] Too many disconnects from streaming API
2024-03-05 19:07:44.755 car_id=1 [info] Creating new connection … 
2024-03-05 19:07:44.755 car_id=1 [info] Disconnecting ...
2024-03-05 19:07:44.768 car_id=1 [info] Connecting ...
2024-03-05 19:14:11.130 [warning] Too many disconnects from streaming API
2024-03-05 19:14:11.130 car_id=1 [info] Creating new connection … 
2024-03-05 19:14:11.130 car_id=1 [info] Disconnecting ...
2024-03-05 19:14:11.144 car_id=1 [info] Connecting ...
2024-03-05 19:20:41.166 [warning] Too many disconnects from streaming API
2024-03-05 19:20:41.167 car_id=1 [info] Creating new connection … 
2024-03-05 19:20:41.167 car_id=1 [info] Disconnecting ...
2024-03-05 19:20:41.177 car_id=1 [info] Connecting ...
2024-03-05 19:27:06.636 [warning] Too many disconnects from streaming API
2024-03-05 19:27:06.637 car_id=1 [info] Creating new connection … 
2024-03-05 19:27:06.637 car_id=1 [info] Disconnecting ...
2024-03-05 19:27:06.650 car_id=1 [info] Connecting ...
2024-03-05 19:33:32.269 [warning] Too many disconnects from streaming API
2024-03-05 19:33:32.269 car_id=1 [info] Creating new connection … 
2024-03-05 19:33:32.269 car_id=1 [info] Disconnecting ...
2024-03-05 19:33:32.283 car_id=1 [info] Connecting ...
2024-03-05 19:36:35.082 [info] Start of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1709696194, heading: 1, latitude: XX.XXXXXX, longitude: XX.XXXXXX, native_latitude: XX.XXXXXX, native_location_supported: 1, native_longitude: XX.XXXXXX, native_type: "wgs", power: 3, shift_state: "R", speed: 0, timestamp: 1709696194766}
2024-03-05 19:36:35.082 car_id=1 [info] Driving / Start
2024-03-05 19:39:58.596 [warning] Too many disconnects from streaming API
2024-03-05 19:39:58.597 car_id=1 [info] Creating new connection … 
2024-03-05 19:39:58.597 car_id=1 [info] Disconnecting ...
2024-03-05 19:39:58.607 car_id=1 [info] Connecting ...
2024-03-05 19:46:29.061 [warning] Too many disconnects from streaming API
2024-03-05 19:46:29.062 car_id=1 [info] Creating new connection … 
2024-03-05 19:46:29.062 car_id=1 [info] Disconnecting ...
2024-03-05 19:46:29.082 car_id=1 [info] Connecting ...
2024-03-05 20:27:50.678 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (270.492 ms)
2024-03-05 20:27:51.094 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (412.728 ms)
2024-03-05 20:27:51.127 [info] End of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1709699269, heading: 279, latitude: XX.XXXXXX, longitude: XX.XXXXXX, native_latitude: XX.XXXXXX, native_location_supported: 1, native_longitude: XX.XXXXXX, native_type: "wgs", power: 1, shift_state: "P", speed: 0, timestamp: 1709699270282}
2024-03-05 20:27:51.127 car_id=1 [info] Driving / Ended / 83 km – 51 min
2024-03-05 20:27:51.128 car_id=1 [info] Start / :online
2024-03-05 20:29:09.701 car_id=1 [info] Received stream data: :inactive
2024-03-05 20:36:15.102 car_id=1 [info] Received stream data: :inactive
2024-03-05 20:36:25.845 [warning] Too many disconnects from streaming API
2024-03-05 20:36:25.845 car_id=1 [info] Creating new connection … 
2024-03-05 20:36:25.845 car_id=1 [info] Disconnecting ...
2024-03-05 20:36:25.862 car_id=1 [info] Connecting ...
2024-03-05 20:47:18.195 [warning] Too many disconnects from streaming API
2024-03-05 20:47:18.195 car_id=1 [info] Creating new connection … 
2024-03-05 20:47:18.196 car_id=1 [info] Disconnecting ...
2024-03-05 20:47:18.211 car_id=1 [info] Connecting ...
2024-03-05 20:47:34.018 car_id=1 [info] Charging detected: -2 kW
2024-03-05 20:47:34.349 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (195.689 ms)
2024-03-05 20:47:34.403 car_id=1 [info] Charging / SOC: 31%
2024-03-05 20:47:34.403 car_id=1 [info] Disconnecting ...
2024-03-05 22:26:10.734 [info] Refreshing access token ...
2024-03-05 22:26:11.048 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (313.362 ms)
2024-03-05 22:26:11.056 [info] Scheduling token refresh in 6 h
2024-03-05 22:26:15.020 [info] Adding elevation to 60 positions ...
2024-03-06 04:26:11.057 [info] Refreshing access token ...
2024-03-06 04:26:11.372 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (314.366 ms)
2024-03-06 04:26:11.381 [info] Scheduling token refresh in 6 h
2024-03-06 04:42:45.668 car_id=1 [info] Derived efficiency factor: 149.0 Wh/km (10x confirmed)
2024-03-06 04:42:45.669 car_id=1 [info] Charging / Complete / 36.52 kWh – 475 min
2024-03-06 04:42:45.675 car_id=1 [info] Start / :online
2024-03-06 04:42:45.701 car_id=1 [info] Connecting ...
2024-03-06 04:45:48.216 car_id=1 [info] Suspending logging
2024-03-06 05:15:48.410 car_id=1 [info] Start / :asleep
2024-03-06 05:15:48.420 car_id=1 [info] Disconnecting ...
2024-03-06 09:11:37.007 car_id=1 [info] Start / :online
2024-03-06 09:11:37.044 car_id=1 [info] Connecting ...
2024-03-06 09:14:39.036 car_id=1 [info] Suspending logging
2024-03-06 09:25:08.563 car_id=1 [info] Fetching vehicle state ...
2024-03-06 09:25:08.847 car_id=1 [info] Start / :asleep
2024-03-06 09:25:08.858 car_id=1 [info] Disconnecting ...
2024-03-06 10:26:11.383 [info] Refreshing access token ...
2024-03-06 10:26:11.839 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (455.390 ms)
2024-03-06 10:26:11.847 [info] Scheduling token refresh in 6 h
```

### Screenshots

Zoomed view of "Drive" showing that the streaming API was not connected for the first ~11 minutes:

<img width="537" alt="Screenshot 2024-03-06 at 12 47 49 PM" src="https://github.com/teslamate-org/teslamate/assets/111825925/7f25f25f-fa1b-4f8e-9b88-43a07cf8f3aa">

### Type of installation

Docker

### Version

v1.28.3

**Comment by JakobLichterfeld** [comment]
TeslaMate assume the vehicle will be going to sleep and therefore suspend active logging. If you start a new drive during this time, it may be lacking. No one fits all solution, the simplest use web access and click on 'cancel sleep' before starting the new drive.

**Comment by TesLocker** [comment]
Thank you!  Is there a way to activate the "cancel sleep" option via automation?  For example, if my garage door just opened, TeslaMate should not assume the vehicle is going to sleep in the next 15 mins.  TeslaMate is accessible via the Home Assistant integration, if that facilities automating the task.

**Comment by cwanja** [comment]
See #3701.
